### PR TITLE
Navigation: use pageNav and subTitle in Dashboards > Settings > Links

### LINKED ISSUE #55116: Page layout: Dashboard settings -> Links -> New link
- [x] add `subTitle`
- [x] add `pageNav` (see `TeamPages`)

### PR DESCRIPTION
**Pull request by eledobleefe**
**What this PR does / why we need it**:
This PR adds `pageNav` and `subTitle` to the Links page.

**Which issue(s) this PR fixes**:

Fixes #55116

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/dashboard/components/DashboardSettings/LinksSettings.tsx [hunk 1]
```diff
@@ -11,24 +13,46 @@ export type LinkSettingsMode = 'list' | 'new' | 'edit';
...
+  const [isNew, setIsNew] = useState<boolean>(false);
+  const { chrome } = useGrafana();
+
+  React.useEffect(() => {
+    if (sectionNav) {
+      chrome.update({
+        sectionNav: sectionNav.node,
+      });
+    }
+  }, [sectionNav, chrome]);
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/LinksSettings.tsx:25]: is this necessary? i think `Page` already does this 🤔

File: public/app/features/dashboard/components/DashboardSettings/LinksSettings.tsx [hunk 2]
```diff
@@ -11,24 +13,46 @@ export type LinkSettingsMode = 'list' | 'new' | 'edit';
...
 
   const isEditing = editIdx !== null;
 
+  let pageNav: NavModelItem | undefined;
+  if (isEditing) {
+    const title = isNew ? 'New link' : 'Edit link';
+    pageNav = {
+      text: title,
+      subTitle: 'Manage a specific link',
+    };
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/LinksSettings.tsx:51]: think we probably want a different subtitle when creating vs when editing

File: public/app/features/dashboard/components/DashboardSettings/LinksSettings.tsx (hunk 1 above)
Comment (eledobleefe) [reply public/app/features/dashboard/components/DashboardSettings/LinksSettings.tsx:25]: Yes, you're right. Before adding the `pageNav` in the `New link` and `Edit link` pages, I saw 'Link' twice in the `Breadcrumbs`and I instantly remember the solution of the `Explore` page. I'll get rid of it. Thanks!

File: public/app/features/dashboard/components/DashboardSettings/LinksSettings.test.tsx [hunk 3]
```diff
@@ -22,9 +23,9 @@ function setup(dashboard: DashboardModel) {
 
   return render(
     <GrafanaContext.Provider value={getGrafanaContextMock()}>
-      <BrowserRouter>
-        <LinksSettings dashboard={dashboard} sectionNav={sectionNav} />
-      </BrowserRouter>
+      <Router history={locationService.getHistory()}>
+        <DashboardSettings editview="links" dashboard={dashboard} sectionNav={sectionNav} pageNav={sectionNav.node} />
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/LinksSettings.test.tsx:27]:
```suggestion
        // Need to use DashboardSettings here as it's responsible for fetching the state back from location
        <DashboardSettings editview="links" dashboard={dashboard} sectionNav={sectionNav} pageNav={sectionNav.node} />
```

**Review by ashharrison90 — APPROVED** [review]
looks great! 🙌

**Pull request merged**